Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/impl/SHA2PasswordService.java

```diff
@@ -1,104 +1,104 @@
- * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
- * DISCLAIMED. IN NO EVENT SHALL THE UNITED STATES GOVERNMENT BE LIABLE FOR ANY
- * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
- * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
- * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
- * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
- * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
- * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
- */
-package gov.hhs.fha.nhinc.admingui.services.impl;
-
-import gov.hhs.fha.nhinc.admingui.services.exception.PasswordServiceException;
-
-import java.io.ByteArrayOutputStream;
-import java.io.IOException;
-import java.security.MessageDigest;
-import java.security.NoSuchAlgorithmException;
-import java.security.SecureRandom;
-
-import org.apache.commons.codec.binary.Base64;
-
-/**
- * The Class SHA2PasswordService.
- *
- * @author msw
- */
-public class SHA2PasswordService extends AbstractBase64EncodedPasswordService {
-
-    public static final String HASH_ALGORITHM = "SHA-512";
-
-    private static final SecureRandom RANDOM = new SecureRandom();
-
-    /**
-     *
-     */
-    public SHA2PasswordService() {
-        super();
-    }
-
-    /*
-     * (non-Javadoc)
-     *
-     * @see gov.hhs.fha.nhinc.admingui.services.PasswordService#calculateHash(byte[])
-     */
-    @Override
-    public byte[] calculateHash(byte[] input) throws PasswordServiceException {
-        byte[] digest;
-
-        try {
-            MessageDigest md = MessageDigest.getInstance(HASH_ALGORITHM);
-            digest = md.digest(input);
-        } catch (NoSuchAlgorithmException e) {
-            throw new PasswordServiceException("Unable to calculate digest for password.", e);
-        }
-
-        return encode(digest);
-    }
-
-    /**
-     *
-     * @return the salt value
-     */
-    @Override
-    public byte[] generateRandomSalt() {
-        byte[] salt = new byte[16];
-        RANDOM.nextBytes(salt);
-        return Base64.encodeBase64(salt);
-    }
-
-    /**
-     *
-     * @param salt the salt value
-     * @param password the password
-     * @return calculated hash as byte array
-     * @throws IOException
-     * @throws PasswordServiceException
-     */
-    @Override
-    public byte[] calculateHash(byte[] salt, byte[] password) throws IOException, PasswordServiceException {
-        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
-
-        outputStream.write(salt);
-        outputStream.write(password);
-
-        return calculateHash(outputStream.toByteArray());
-    }
-}
+/*
+ * Copyright (c) 2009-2017, United States Government, as represented by the Secretary of Health and Human Services.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
```

Comment:
Why this file change a lot?  Undo this file and manual change to 2009-2017 in notepad++ to reserve this format.